src/resolver.ts

```diff
@@ -1,15 +1,34 @@
 import { UserInput } from './UserInput';
 import { User } from './entity/User';
 import { AppDataSource } from './data-source';
+import { CustomError } from './custom-errror';
 
 export const resolvers = {
   Mutation: {
     createUser: async (_, args: { data: UserInput }) => {
       const { data } = args;
+      if (data.password.length < 6) {
+        throw new CustomError('A senha deve conter pelo menos 6 caracteres', 400);
+      }
+      if (!isThereALetter(data.password)) {
+        throw new CustomError('A senha deve conter pelo menos 1 letra', 400);
+      }
+      if (isThereANumber(data.password)) {
+        throw new CustomError('A senha deve conter pelo menos 1 número', 400);
+      }
+      const hasEmail = await isEmailAlreadyUsed(data.email);
+      if (hasEmail) {
+        throw new CustomError('Email já utilizado', 400);
+      }
       console.log('Inserting a new user into the database...');
       const user = new User();
       user.name = data.name;
-      user.password = data.password;
+      user.hash = 'test';
+      const { scrypt } = await import('node:crypto');
+      scrypt(data.password, 'salt', 10, (err, derivedKey) => {
+        if (err) throw new CustomError('Erro Interno', 401, 'erro no algoritmo hash');
+        user.hash = derivedKey.toString('hex');
+      });
       user.email = data.email;
```

Comment:
Aqui tem algo arriscado que é usar o callback e seguir com a opração na linha 32. Se o script de hash demorar mais do que o restante da operação, esse callback acaba sendo executado depois da chamada já ter sido retornada para o client. Aqui, dado que vc já está usando a sintaxe async/await no createUser, eu recomendaria vc transofrmar a operação de hash em Promise. Algo comum que fazemos quando alguma função externa só tem callback.

Alguns exemplos [nesse link](https://zellwk.com/blog/converting-callbacks-to-promises/).